src/creator.js

```diff
@@ -0,0 +1,1469 @@
+		term.setAttribute('data-x', ui.position.left)
+		term.setAttribute('data-y', ui.position.top)
+
+		return _drawBoard()
+	}
+
+	// called from Materia creator page
+	const onSaveClicked = function (mode) {
+		if (mode == null) { mode = 'save' }
+
+		return !_buildSaveData()
+			? Materia.CreatorCore.cancelSave('Widget needs a title and at least one term.')
+			: Materia.CreatorCore.save(_title, _qset)
+	}
+
+	const onSaveComplete = (title, widget, qset, version) => true
+
+	// called from Materia creator page
+	// place the questions in an arbitrary location to be moved
+	const onQuestionImportComplete = items => Array.from(items).map((item) =>
+		_makeTerm(150, 300, item.questions[0].text, null, null, item.id))
+
+	// generate the qset
+	var _buildSaveData = function () {
+		if ((_qset == null)) { _qset = {} }
+		if ((_qset.options == null)) { _qset.options = {} }
+
+		const words = []
+
+		_qset.assets = []
+		_qset.rand = false
+		_qset.name = ''
+		_title = document.getElementById('title').innerHTML
+		let _okToSave = (_title != null) && (_title !== '') ? true : false
+
+		const items = []
+		const dots = document.querySelectorAll('.term')
+		for (let dot of Array.from(dots)) {
+			const item = {}
+			const label = dot.childNodes[0].innerHTML
+
+			const answer = {
+				text: label,
+				value: 100,
+				id: ''
+			}
+			item.answers = [answer]
+			item.assets = []
+
+			const question = { text: label }
+			item.questions = [question]
+			item.type = 'QA'
+			item.id = dot.getAttribute('data-id') || ''
+
+			if (flag3D === false) {
+				item.options = {
+					labelBoxX: parseInt(dot.style.left.replace('px', '')),
+					labelBoxY: parseInt(dot.style.top.replace('px', '')),
+					endPointX: parseInt(dot.getAttribute('data-x')),
+					endPointY: parseInt(dot.getAttribute('data-y'))
+				}
+
+			} else {
+				let vertex
+				listOfVertex.forEach((element) => {
+					if (element.dataTermID == dot.id) { vertex = element }
+				})
+
+				// The sphere name is generated based on the dotID automatically
+				item.options = {
+					labelBoxX: parseInt(dot.style.left.replace('px', '')),
+					labelBoxY: parseInt(dot.style.top.replace('px', '')),
+					endPointX: parseInt(dot.getAttribute('data-x')),
+					endPointY: parseInt(dot.getAttribute('data-y')),
+					vertex: {
+						dataTermID: vertex.dataTermID,
+						dotID: vertex.dotID,
+						faceIndex: vertex.faceIndex,
+						point: vertex.point,
+						uv: vertex.uv,
+						// sphereRadius: sphereRadius,
+					}
+				}
+			}
+
+			items.push(item)
+		}
+
+		_qset.items = items
+
+		if (items.length < 1) { _okToSave = false }
+
+		if (_anchorOpacity.indexOf('frosted') > -1) { _anchorOpacityValue = 0.5 }
+		else if (_anchorOpacity.indexOf('transparent') > -1) { anchorOpacityValue = 0.0 }
+
+		_qset.options.flag3D === true ? qsetOption3D(_qset)
+			: flag3D === true ? qsetOption3D(_qset)
+				: qsetOption(_qset)
+
+		_qset.version = "2"
+		return _okToSave
```

Comment:
Return not required

Reply:
Removing the return causes the widget to not be able to save as a draft.